react-components/src/architecture/base/commands/BaseCommand.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+/* eslint-disable @typescript-eslint/class-literal-property-style */
+
+import { clear, remove } from '../utilities/extensions/arrayExtensions';
+
+type UpdateDelegate = () => void;
+
+export type Tooltip = {
+  key: string;
+  fallback: string;
+};
+
+export abstract class BaseCommand {
+  // ==================================================
+  // INSTANCE FIELDS
+  // ==================================================
+
+  private readonly _listeners: UpdateDelegate[] = [];
+
+  // ==================================================
+  // VIRTUAL METHODS (To be override)
+  // =================================================
+
+  public get name(): string {
+    return this.tooltip.fallback;
+  }
+
+  public get shortCutKey(): string | undefined {
+    return undefined;
+  }
+
+  public get tooltip(): Tooltip {
+    return { key: '', fallback: '' };
+  }
+
+  public get icon(): string {
+    return '';
+  }
```

Comment:
I see that some of these properties are very tied to the command being used as a button, but if I understand correctly, many commands will probably not be tied to a button? Would it make sense to make "button-commands" a separate (sub-)class?

Reply:
If you don not use it as buttons, you don't need to implement the tooltip and the icon. This has never been a problem actually. If you make a base class for  these, you will loose a lot of flexibility.